### PR TITLE
SOC-3211 Mark deleted and reported posts in discussions tag

### PR DESCRIPTION
**Pull request by perjg**
@Grunny @garthwebb

### CODE REVIEW

File: extensions/wikia/EmbeddableDiscussions/scripts/templates.mustache.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
 /* jshint ignore:start */ define( 'embeddablediscussions.templates.mustache', [], function() { 'use strict'; return {
```
Comment (WikiaTech) [inline extensions/wikia/EmbeddableDiscussions/scripts/templates.mustache.js:1]: MINOR Reformat the code to have only one statement per line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 2 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ext.embeddableDiscussions.js#L10](https://github.com/Wikia/app/blob/6f342ba6d9f8a9a90dfbbd1b66be34dba609c743/extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js#L10): Function has 8 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. MAJOR [ext.embeddableDiscussions.js#L69](https://github.com/Wikia/app/blob/6f342ba6d9f8a9a90dfbbd1b66be34dba609c743/extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js#L69): Restrict what this loop acts on by testing each property. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AForIn)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 2 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ext.embeddableDiscussions.js#L10](https://github.com/Wikia/app/blob/c05d878019431b81441c7e43c815d2269162353a/extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js#L10): Function has 8 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. MAJOR [ext.embeddableDiscussions.js#L69](https://github.com/Wikia/app/blob/c05d878019431b81441c7e43c815d2269162353a/extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js#L69): Restrict what this loop acts on by testing each property. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AForIn)

File: extensions/wikia/EmbeddableDiscussions/styles/EmbeddableDiscussions.scss [hunk 2]
```diff
@@ -28,6 +28,11 @@ $font-size-small: 12px;
 $font-size-regular: 14px;
 $font-size-large: 16px;
 
+$post-mark-icon-size: 12px;
+$discussion-deleted-mark-color: #ed1c24;
+$discussion-reported-mark-color: #f08a24;
+$discussion-deleted-actions-opacity: 0.5;
```
Comment (Grunny) [inline extensions/wikia/EmbeddableDiscussions/styles/EmbeddableDiscussions.scss]: Remove the leading 0 per conventions.

File: extensions/wikia/EmbeddableDiscussions/styles/EmbeddableDiscussions.scss [hunk 3]
```diff
@@ -85,6 +90,10 @@ $font-size-large: 16px;
 	}
 
 	.embeddable-discussions-content {
+		&.is-deleted {
```
Comment (Grunny) [inline extensions/wikia/EmbeddableDiscussions/styles/EmbeddableDiscussions.scss:100]: This nested rule should be below the `font-size` property per conventions.

File: extensions/wikia/EmbeddableDiscussions/styles/EmbeddableDiscussions.scss [hunk 4]
```diff
@@ -120,6 +129,10 @@ $font-size-large: 16px;
 	}
 
 	.embeddable-discussions-post-actions li {
+		&.is-deleted {
```
Comment (Grunny) [inline extensions/wikia/EmbeddableDiscussions/styles/EmbeddableDiscussions.scss]: This nested rule should be below the properties below per conventions.

File: extensions/wikia/EmbeddableDiscussions/styles/EmbeddableDiscussions.scss [hunk 5]
```diff
@@ -144,9 +157,35 @@ $font-size-large: 16px;
 	}
 
 	.embeddable-discussions-post-detail {
+		&.is-deleted,
+		&.is-reported {
+			&::before {
+				border-color: transparent;
+				border-style: solid;
+				border-width: $post-mark-icon-size $post-mark-icon-size 0 0;
+				content: "";
```
Comment (Grunny) [inline extensions/wikia/EmbeddableDiscussions/styles/EmbeddableDiscussions.scss]: Should use single quotes per conventions.

File: extensions/wikia/EmbeddableDiscussions/styles/EmbeddableDiscussions.scss [hunk 6]
```diff
@@ -144,9 +157,35 @@ $font-size-large: 16px;
...
+		&.is-reported {
+			&::before {
+				border-top-color: $discussion-reported-mark-color;
+			}
+		}
+
 		border: 1px solid $card-border-color;
 		margin: 0 0 20px;
 		padding: 20px 0 0;
+		position: relative;
```
Comment (Grunny) [inline extensions/wikia/EmbeddableDiscussions/styles/EmbeddableDiscussions.scss]: These properties should be above the nested rules above.

File: extensions/wikia/EmbeddableDiscussions/styles/EmbeddableDiscussions.scss [hunk 7]
```diff
@@ -207,6 +246,10 @@ $font-size-large: 16px;
 	}
 
 	.embeddable-discussions-title {
+		&.is-deleted {
```
Comment (Grunny) [inline extensions/wikia/EmbeddableDiscussions/styles/EmbeddableDiscussions.scss]: This nested rule should be below the properties below.

**Review by Grunny — CHANGES_REQUESTED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 2 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ext.embeddableDiscussions.js#L10](https://github.com/Wikia/app/blob/597b7d3f856315e67ec00acfdcfc8682fd74a43d/extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js#L10): Function has 8 parameters which is greater than 7 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. MAJOR [ext.embeddableDiscussions.js#L69](https://github.com/Wikia/app/blob/597b7d3f856315e67ec00acfdcfc8682fd74a43d/extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js#L69): Restrict what this loop acts on by testing each property. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AForIn)

**Comment by perjg** [comment]
@Grunny I've fixed the style guide issues in the .scss file.

**Review by Grunny — APPROVED** [review]
LGTM.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 2 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ext.embeddableDiscussions.js#L10](https://github.com/Wikia/app/blob/a46bd74c6be249f39c6bf685f1835ed0b33e5b7a/extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js#L10): Function has 8 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. MAJOR [ext.embeddableDiscussions.js#L70](https://github.com/Wikia/app/blob/a46bd74c6be249f39c6bf685f1835ed0b33e5b7a/extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js#L70): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 2 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ext.embeddableDiscussions.js#L10](https://github.com/Wikia/app/blob/2dca5dd61c505e03334b163d4b991ae7040f0b51/extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js#L10): Function has 8 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. MAJOR [ext.embeddableDiscussions.js#L70](https://github.com/Wikia/app/blob/2dca5dd61c505e03334b163d4b991ae7040f0b51/extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js#L70): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)

**Pull request merged**